serving/src/test/java/feast/serving/it/ServingServiceSnowflakeStorageIT.java

```diff
@@ -0,0 +1,208 @@
+import io.grpc.StatusRuntimeException;
+import java.io.File;
+import java.io.IOException;
+import java.net.UnknownHostException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.runners.model.InitializationError;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.kafka.core.DefaultKafkaProducerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@ActiveProfiles("it")
+@SpringBootTest()
+@Testcontainers
+public class ServingServiceSnowflakeStorageIT {
+
+  static final String CORE = "core_1";
+  static final int CORE_START_MAX_WAIT_TIME_IN_MINUTES = 3;
+  static final int FEAST_CORE_PORT = 6565;
+  static final int FEAST_SERVING_PORT = 6566;
+  static final int MAX_AGE_SECOND = 30;
+  static final String PROJECT_NAME = "myproject4";
+  static final String FEATURE_SET = "feature_set";
+  static final String ENTITY_NAME = "ENTITY_ID_PRIMARY";
+  static final String TEST_FEATURE_1 = "FEATURE_1";
+
+  @DynamicPropertySource
+  static void initialize(DynamicPropertyRegistry registry) throws UnknownHostException {}
+
+  @ClassRule @Container
+  public static DockerComposeContainer environment =
+      new DockerComposeContainer(
+              new File("src/test/resources/docker-compose/docker-compose-it-core.yml"))
+          .withExposedService(
+              CORE,
+              6565,
+              Wait.forLogMessage(".*gRPC Server started.*\\n", 1)
+                  .withStartupTimeout(Duration.ofMinutes(CORE_START_MAX_WAIT_TIME_IN_MINUTES)));
+
+  @BeforeAll
+  static void globalSetup() throws IOException, InitializationError, InterruptedException {
+    System.out.print("global setup");
+  }
+
+  public KafkaTemplate<String, FeatureRow> specKafkaTemplate() {
+
+    Map<String, Object> props = new HashMap<>();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "kafka:9092,localhost:9094");
+
+    KafkaTemplate<String, FeatureRow> t =
+        new KafkaTemplate<>(
+            new DefaultKafkaProducerFactory<>(
+                props, new StringSerializer(), new KafkaSerialization.ProtoSerializer<>()));
+    t.setDefaultTopic("feast-features");
```

Comment:
topic is used as multiple place .. could be defined as constant variable

Reply:
Done